microservices/first/utils/db_manager.py

```diff
@@ -0,0 +1,134 @@
+    c.execute(
+              '''CREATE TABLE IF NOT EXISTS deployments (
+              id INTEGER PRIMARY KEY, 
+              leader_ID INTEGER UNIQUE
+              repo_name TEXT, 
+              repo_branch TEXT,
+              status TEXT DEFAULT 'STARTED')'''
+            )
+
+  def create_results_table(self) -> None:
+    """Create the results table in the database."""
+    c = self.conn.cursor()
+    c.execute(
+        '''CREATE TABLE IF NOT EXISTS results (
+            id INTEGER PRIMARY KEY,
+            deployment_id INTEGER,
+            result TEXT,
+            FOREIGN KEY(deployment_id) REFERENCES deployments(id)
+        )'''
+    )
+
+  def add_result(self, deployment_id: int, result: str) -> int:
+    """
+    Add a new result to a specific deployment.
+
+    Returns: Row number of result.
+    """
+    sql = '''INSERT INTO results(deployment_id, result) VALUES(?, ?)'''
+    cur = self.conn.cursor()
+    cur.execute(sql, (deployment_id, result))
+    self.conn.commit()
+    return cur.lastrowid
+  
+  def add_results(self, deployment_id: int, results: list) -> None:
+    """Add multiple results to a specific deployment."""
+    sql = '''INSERT INTO results(deployment_id, result) VALUES(?, ?)'''
+    cur = self.conn.cursor()
+    for result in results:
+        cur.execute(sql, (deployment_id, result))
+    self.conn.commit()
+
+  def get_results(self, deployment_id: int) -> list[str]:
+    """Retrieve all results for a specific deployment."""
+    cur = self.conn.cursor()
+    cur.execute("SELECT result FROM results WHERE deployment_id=?", (deployment_id,))
+    results = [row[0] for row in cur.fetchall()]
+    return results
+
+  def add_deployment(self, leader_id: str, repo_name: str, repo_branch: str) -> int:
+    """
+    Add a new deployment to the database.
+    
+    param leader_id: The leader's ID
+    param repo_name: The name of the repository for the deployment.
+    param repo_branch: The branch of the repository for the deployment.
+    """
+    sql = '''INSERT INTO deployments(leader_id, repo_name, repo_branch) VALUES(?,?,?)'''
+    cur = self.conn.cursor()
+    cur.execute(sql, (int(leader_id), repo_name, repo_branch))
+    self.conn.commit()
+    return cur.lastrowid
+
+  def update_deployment_fields(self, deployment_id: int, updates: dict[str, str]) -> None:
+    """
+    Update specified fields of an existing deployment.
+
+    :param deployment_id: The ID of the deployment to update.
+    :param updates: A dictionary where keys are column names and values are the new values for those columns.
+    """
+    parameters = [f"{key} = ?" for key in updates.keys()]
+    sql = f"UPDATE deployments SET {', '.join(parameters)} WHERE id = ?"
+    values = list(updates.values()) + [deployment_id]
+    
+    cur = self.conn.cursor()
+    cur.execute(sql, values)
+    self.conn.commit()
+
+  def get_deployment(self, deployment_id: int) -> Dict[str, Union[str, int, deployment_status.DeploymentStatus]]:
```

Comment:
The output of this seems to include `deployment_status` yet nowhere is the leader output being stored as this enum? It does not seem like this enum value is ever stored or interpreted from the leader to `get_deployment` lifecycle.

Reply:
ok